### DISCUSSION TITLE
TransactionContext Error : Conflict on update!

### DISCUSSION THREAD

**Discussion by Stongtong** [duckdb/duckdb#6949] [General]
hi guys，when I update data, I got this problem 
**TransactionContext Error : Conflict on update!**, should I add synchonized key word on my update method?

**Comment by Tishj** [comment]
I think we need some more context on what caused this issue :)

**Comment by Stongtong** [comment]
the error detail is blow:
\#\#\# Error updating database. Cause: java.sql.SQLException: TransactionContext Error: Conflict on update! 
\#\#\# The error may exist in mapper/duck/cache/XXXXX.java (best guess) 
\#\#\# The error may involve mapper.duck.cache.XXXXX.updateCount-Inline 
\#\#\# The error occurred while setting parameters 
\#\#\# SQL: update XXXXX set cache_size = ? where cache_id = ? 
\#\#\# Cause: java.sql.SQLException: TransactionContext Error: Conflict on update! ; uncategorized SQLException; SQL state [null]; error code [0]; TransactionContext Error: Conflict on update!; nested exception is java.sql.SQLException: TransactionContext Error: Conflict on update!

**Comment by Stongtong** [comment]
Multi thread import parquet file to duckdb (one table imported has more par file),  and when a par file import finished, program would update number of record in summarize table on duckdb, the exception throw on update number process
